doc/atdj.rst

```diff
@@ -0,0 +1,319 @@
+***************************
+Java/Json support with atdj
+***************************
+
+The ATDJ tool generates a Java interface from an ATD interface. In
+particular, given a set of ATD types, this tool generates a set of Java
+classes representing those types. These classes may then be instantiated
+from JSON representations of those same ATD types.
+
+The primary benefits of using the generated interface, over manually
+manipulating JSON strings from within Java, are safety and ease of use.
+Specifically, the generated interface offers the following features:
+
+-  JSON strings are automatically checked for correctness with respect
+   to the ATD specificion.
+
+-  Details such as optional fields and their associated default values
+   are automatically handled.
+
+-  Several utility methods are included “for free”. These support
+   equality testing, the visitor pattern and conversion back to JSON.
+
+Installation
+============
+
+Build and install the ``atdj`` command with `opam <https://opam.ocaml.org/>`__:
+
+  ::
+
+       opam install atdj
+
+Quick-start
+===========
+
+In this section we briefly describe how to to generate a Java interface
+from an example ATD file ``test.atd``. We then show how to build and run
+an example application ``AtdjTest`` that uses the generated interface.
+
+#. Generate and compile the interface:
+
+   ::
+
+       ./atdj -graph -package com.mylife.test test.atd
+       export CLASSPATH='.:json.jar'
+       javac com/mylife/test/*.java
+
+#. Compile and run the example, saving the output for later inspection:
+
+   ::
+
+       javac AtdjTest.java
+       java AtdjTest >test.out
+
+#. Optionally, generate Javadoc documentation:
+
+   ::
+
+         javadoc -d doc -public com.mylife.test
+
+   The resulting documentation is located in the directory ``doc``.
+
+#. Optionally, generate a class graph of the generated interface:
+
+   ::
+
+       dot -Tpdf test.dot >test.pdf
+
+The output file ``test.pdf`` contains a class graph of the generated
+Java interface. The required ``dot`` program is part of the Graphviz
+graph visualisation package, and may be downloaded from
+http://www.graphviz.org/.
+
+In the following sections we discuss the individual steps in more
+detail, using the example from above.
+
+Generating the interface
+========================
+
+In this section we describe the process of generating a Java interface
+from an ATD specification.
+
+A Java interface is generated from an ATD file as
+
+::
+
+    ./atdj -package <package> <atd_file>
```

Comment:
A user is unlikely to have atdj in their current working directory, so i'd change this to `atdj` instead.

Reply:
fixed that.